File: Easy leetcode/2591. Distribute Money to Maximum Children.py
class Solution:
    def distMoney(self, money: int, children: int) -> int:
        if money < children:
            return -1
        if money == children * 8:
            return children
        if money > children * 8:
            return children - 1

        money -= children
        ans = money // 7
        rem = money % 7
        children -= ans

        if rem == 3 and children == 1:
            ans -= 1
        return ans

File: Easy leetcode/2591. Distribute Money to Maximum Children.java
class Solution {
    public int distMoney(int money, int children) {
        if (money < children) {
            return -1;
        }
        if (money == children * 8) {
            return children;
        }
        if (money > children * 8) {
            return children - 1;
        }

        money -= children;
        int ans = money / 7;
        int rem = money % 7;
        children -= ans;

        if (rem == 3 && children == 1) {
            ans--;
        }
        return ans;
    }
}
